Closes #129. Clarifying the use of wallet string address

dkgaraujo opened the pull request:
Signed-off-by: Douglas Araujo <dkg.araujo@gmail.com>

dkgaraujo committed:
Closes #129. Clarifying the use of wallet string address

Signed-off-by: Douglas Araujo <dkg.araujo@gmail.com>

Update README.md to clarify new wallet's address

Co-authored-by: Sam Stuewe <Samuel.Stuewe@gmail.com>
Signed-off-by: Douglas Araujo <dkg.araujo@gmail.com>

Simplified text along suggestion

Signed-off-by: Douglas Araujo <dkg.araujo@gmail.com>

dkgaraujo committed:
Closes #129. Clarifying the use of wallet string address

Signed-off-by: Douglas Araujo <dkg.araujo@gmail.com>

HalosGhost commented on README.md:
````diff
@@ -150,7 +150,9 @@ In each of the below commands, you should pass `atomizer-compose.cfg` instead of
   ```
 
-* Send currency from one wallet to another (e.g., 30 atomic units of currency)
+  Note the resulting string ("usd1qrw038lx5n4wxx3yvuwdndpr7gnm347d6pn37uywgudzq90w7fsuk52kd5u"). It is the new wallet's address, and will be used in the following step.
````
```suggestion
  Note the resulting string ("usd1qrw038lx5n4wxx3yvuwdndpr7gnm347d6pn37uywgudzq90w7fsuk52kd5u").
  It is the new wallet's address, and will be used in the following step.
```

HalosGhost commented on README.md:
```diff
@@ -150,7 +150,9 @@ In each of the below commands, you should pass `atomizer-compose.cfg` instead of
-* Send currency from one wallet to another (e.g., 30 atomic units of currency)
+  Note the resulting string ("usd1qrw038lx5n4wxx3yvuwdndpr7gnm347d6pn37uywgudzq90w7fsuk52kd5u"). It is the new wallet's address, and will be used in the following step.
+
+* Send currency from one wallet to another (e.g., 30 atomic units of currency) **Note that you may need to change the wallet string in the code below ("usd1qrw0...") depending on the result of the previous step.**
```
```suggestion
* Send currency from one wallet to another (e.g., 30 atomic units of currency).
  **Note that you will need to change the wallet string in the code below ("usd1qrw0...") depending on the result of the previous step.**
```

HalosGhost commented on README.md:
````diff
@@ -171,7 +173,7 @@ In each of the below commands, you should pass `atomizer-compose.cfg` instead of
   ```
 
-* Import coins to the receiving wallet
+* Import coins to the receiving wallet. Note that the string after `importinput` is the same as the result from step above of sending the currency between wallets.
````
```suggestion
* Import coins to the receiving wallet.
  Note that the string after `importinput` is the same as the result from step above of sending the currency between wallets.
```

HalosGhost reviewed (changes requested):
I think this kind of clarity is good, but I do have a couple of requests.

I made a few suggestions inline to keep one-sentence-per-line (a gift for reviewers of diffs); but I also am a bit worried that some of the specificity might actually be a bit too rigid.

This walkthrough is meant to get people interacting with the system quickly, but this is not a production system, doesn't follow semver, and the CLI (what little of it there is) is purely for testing and demonstration. To that end, I wonder if it might make more sense (from a documentation-burden standpoint) to have more general statements (e.g., “send currency from the first wallet to the address of the second wallet created in the previous step”) which are more likely to remain evergreen.

Thoughts?

dkgaraujo committed:
Update README.md to clarify new wallet's address

Co-authored-by: Sam Stuewe <Samuel.Stuewe@gmail.com>
Signed-off-by: Douglas Araujo <dkg.araujo@gmail.com>

dkgaraujo committed:
Simplified text along suggestion

Signed-off-by: Douglas Araujo <dkg.araujo@gmail.com>

dkgaraujo commented on README.md:
````diff
@@ -174,7 +171,7 @@ In each of the below commands, you should pass `atomizer-compose.cfg` instead of
   ```
 
-* Import coins to the receiving wallet. Note that the string after `importinput` is the same as the result from step above of sending the currency between wallets.
+* Import coins to the receiving wallet using the string after `importinput` from step above.
````
I tried to follow a similar one-line style for this string as well.

dkgaraujo reviewed (commented):
Thank you for the suggestion to further simplify. It is much better indeed. I hope this new version can address that.

dkgaraujo commented:
FYI, tThe recent commits are actually a rebase because one of them was not signed off. The content is the same.

dkgaraujo commented:
Hi @HalosGhost, when you have some time, please feel free to let me know if there is anything outstanding from my side in this PR. I can't seem to find anything else outstanding after I pushed the simpler version.

HalosGhost commented:
@dkgaraujo I'm sorry about the delay! My last week or two has been quite full. I'm making time to go through all the open PRs in the next few days!

dkgaraujo commented:
> @dkgaraujo I'm sorry about the delay! My last week or two has been quite full. I'm making time to go through all the open PRs in the next few days!

No issue - didn't mean to press, I was just checking in case I forgot anything. Happy to proceed at your pace.

HalosGhost reviewed (approved):
Simple clarity improvement to our README walkthrough.

Looks great!

HalosGhost commented:
@dkgaraujo Whenever you're ready, if you can squash these to a single commit, I'll happily merge!

dkgaraujo committed:
Merge branch 'trunk' of github.com:dkgaraujo/opencbdc-tx into trunk
Squashing of previous commits in PR to clarify usage instructions

HalosGhost commented:
@dkgaraujo it looks like you merged in from `trunk`, but didn't rebase/squash. The easiest way to get yourself back to a good spot is probably (assuming you have this fork configured as a remote named `upstream` on your clone) `git pull --rebase upstream trunk`; after that point, if you still need to squash, you can do an interactive rebase.

HalosGhost commented:
@dkgaraujo let me know if you need help squashing.

dkgaraujo commented:
> @dkgaraujo let me know if you need help squashing.

@HalosGhost sorry for the delay - I had to prioritise another project. I'll come back to this by tomorrow at the latest. Thanks for the patience.

HalosGhost commented:
> sorry for the delay

No worries; I'll still be here to review when you're ready!

dkgaraujo commented:
Hi @HalosGhost, thanks again for the patience. I successfully rebased and squashed (I believe) - but in the process of interactively rebasing I noted that I went a couple of steps back from my own first commit, and when I removed the previous commits by other contributors from the interactive page, it effectively deleted them from my version of trunk. So please don't merge it. I think going back and making sure all is there would be an additional complication on this already complicated PR. Maybe we can just cancel this one and reopen another one with a simple, single commit implementing the changes? Happy to hear your views.
